api-1.10/src/main/java/org/openmrs/module/reporting/cohort/definition/DrugOrderCohortDefinition.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.reporting.cohort.definition;
+
+import org.openmrs.Concept;
+import org.openmrs.CareSetting;
+
+import org.openmrs.module.reporting.common.Localized;
+import org.openmrs.module.reporting.definition.configuration.ConfigurationProperty;
+import org.openmrs.module.reporting.definition.configuration.ConfigurationPropertyCachingStrategy;
+import org.openmrs.module.reporting.evaluation.caching.Caching;
+
+import java.util.Date;
+import java.util.List;
+
+@Caching(strategy = ConfigurationPropertyCachingStrategy.class)
+@Localized("reporting.EncounterCohortDefinition")
+public class DrugOrderCohortDefinition extends BaseCohortDefinition {
+
+    public static final long serialVersionUID = 1L;
+
+    @ConfigurationProperty(group = "which")
+    private String which;
+
+    @ConfigurationProperty(value = "drugConcepts")
+    private List<Concept> drugConcepts;
+
+    @ConfigurationProperty(value = "drugSets")
+    private List<Concept> drugSets;
+
+    @ConfigurationProperty(value = "activatedOnOrBefore")
+    private Date activatedOnOrBefore;
+
+    @ConfigurationProperty(value = "activatedOnOrAfter")
+    private Date activatedOnOrAfter;
+
```

Comment:
For the three properties below - activeWithinLastMonths, activeWithinLastDays, and onlyCurrentlyActive, we tend to avoid properties like this as they relate only to the date/time that they are executed.  It would be better to use an actual Date that these refer to, which can be manipulated as needed to get information about any point in time.  I would change these to:
* Date activeOnDate;
* Date activeOnOrAfter;
* Date activeOnOrBefore;

Reply:
I implemented the above as requested.  I would like to however clear my mind on the behavior of `activeOnDate` parameter, whether or not in this case the `dateActivated` and `dateStopped` have to equal this parameter.